quil-rs/src/program/error/mod.rs

```diff
@@ -50,50 +45,44 @@ where
     }
 }
 
-impl<T> From<ParseError> for ProgramError<T> {
+impl<T> From<ParseError> for ProgramParsingError<T> {
     fn from(e: ParseError) -> Self {
         Self::Syntax(SyntaxError::from(e))
     }
 }
 
-impl<T> From<LeftoverError<T>> for ProgramError<T> {
+impl<T> From<LeftoverError<T>> for ProgramParsingError<T> {
     fn from(err: LeftoverError<T>) -> Self {
         Self::Syntax(SyntaxError::from(err))
     }
 }
 
-impl<T> From<SyntaxError<T>> for ProgramError<T> {
+impl<T> From<SyntaxError<T>> for ProgramParsingError<T> {
     fn from(err: SyntaxError<T>) -> Self {
         Self::Syntax(err)
     }
 }
 
-impl<T> ProgramError<T> {
+impl<T> ProgramParsingError<T> {
     /// Convert the parsed output into another type.
     ///
     /// This delegates to [`LeftoverError::map_parsed`] when a [`ProgramError::Leftover`] and does
```

Comment:
This should probably be changed to the new name.

Reply:
The docstring doesn't quite make sense to me. I think it drifted from the actual implementation as there is no `Leftover` variant. I decided to remove that line entirely.